src/Umbraco.Web/Routing/Domain.cs

```diff
@@ -61,5 +78,13 @@ protected Domain(Domain domain)
         /// Gets a value indicating whether the domain is a wildcard domain.
         /// </summary>
         public bool IsWildcard { get; }
+
+        /// <summary>
+        /// Gets the sort order.
+        /// </summary>
+        /// <value>
+        /// The sort order.
+        /// </value>
+        public int SortOrder { get; }
```

Comment:
The `Domain` model (used in `IDomainCache`) also requires a sort order, as the cache doesn't keep the order of the collections.